lib/azure/armrest/configuration.rb

```diff
@@ -0,0 +1,134 @@
+module Azure
+  module Armrest
+    class Configuration
+      # Used to store unique token information.
+      @@tokens = Hash.new([])
+
+      # The client ID used to gather token information.
+      attr_accessor :client_id
+
+      # The client key used to gather token information.
+      attr_accessor :client_key
+
+      # The tenant ID used to gather token information.
+      attr_accessor :tenant_id
+
+      # The subscription ID used in requests and to gather token information.
+      attr_accessor :subscription_id
+
+      # The resource group used for a given http request.
+      attr_accessor :resource_group
+
+      # The API version used for a given http request. The default is 2015-01-01.
+      attr_accessor :api_version
+
+      # The grant type. The default is client_credentials
+      attr_accessor :grant_type
+
+      # The content type specified for http requests. The default is 'application/json'
+      attr_accessor :content_type
+
+      # The accept type specified for http request results. The default is 'application/json'
+      attr_accessor :accept
+
+      # Set the time in which the token expires.
+      attr_writer :token_expiration
+
+      # Yields a new Azure::Armrest::Configuration objects. Note that you must
+      # specify a client_id and client_key. All other parameters are optional.
+      #
+      # Example:
+      #
+      #   config = Azure::Armrest::Configuration.new(
+      #     :client_id  => 'xxxx',
+      #     :client_key => 'yyyy',
+      #     :tenant_id  => 'zzzz'
+      #   )
+      #
+      def initialize(hash)
+        hash.each{ |key,value| send("#{key}=", value) }
+
+        unless client_id && client_key
+          raise ArgumentError, "client_id and client_key must both be specified"
+        end
+
+        # Set defaults if not already set
+        @accept       ||= 'application/json'
+        @content_type ||= 'application/json'
+        @grant_type   ||= 'client_credentials'
+        @api_version  ||= '2015-01-01'
+      end
+
+      # Returns the tokens class variable hash.
+      def tokens
```

Comment:
We really don't need this method as the hash has no value to user. Exposing it actually incurs dangers of being unintentionally overwritten. Besides this kind of method should be a class method.